stablehlo/tests/interpret_case.mlir

```diff
@@ -0,0 +1,46 @@
+// RUN: stablehlo-interpreter --interpret -split-input-file %s | FileCheck %s
+
+// CHECK-LABEL: Evaluated results of function: case
+func.func @case() -> (tensor<2xi64>, tensor<2xi64>) {
+  %index = stablehlo.constant dense<1> : tensor<i32>
+  %result_branch0 = stablehlo.constant dense<0> : tensor<2xi64>
+  %result_branch1 = stablehlo.constant dense<1> : tensor<2xi64>
+  %result0, %result1 = "stablehlo.case"(%index) ({
+    "stablehlo.return"(%result_branch0, %result_branch0) : (tensor<2xi64>, tensor<2xi64>) -> ()
+  }, {
+    "stablehlo.return"(%result_branch1, %result_branch1) : (tensor<2xi64>, tensor<2xi64>) -> ()
+  }, {
+    "stablehlo.return"(%result_branch0, %result_branch0) : (tensor<2xi64>, tensor<2xi64>) -> ()
```

Comment:
We can one two two regions one default and other non-default. This avoids the mix of values.

Reply:
And below we can use the same example using an OOB index value to pick the default branch.